src/ucs/datastruct/bitmap.c

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2023. ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "bitmap.h"
+
+
+static UCS_F_ALWAYS_INLINE size_t
+ucs_bitmap_bits_fns_inline(const ucs_bitmap_word_t *bits, size_t num_words,
+                           size_t start_index, size_t n)
+{
+    size_t word_index = start_index / UCS_BITMAP_BITS_IN_WORD;
+    size_t mask       = ~(ucs_bitmap_word_bit_mask(start_index) - 1);
+    ucs_bitmap_word_t masked_word;
+    unsigned bit_index;
+
+    UCS_BITMAP_CHECK_INDEX(start_index, num_words, <=);
+    while (word_index < num_words) {
+        masked_word = bits[word_index] & mask;
+        while (masked_word != 0) {
+            /* TODO optimize by skipping multiple bits */
```

Comment:
Maybe impl it before merge?
Having nested loop for a single pass over all bits is a bit weird..